cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/Scheduler.java

```diff
@@ -53,8 +58,10 @@
    * @param programType type of program.
    * @return list of Scheduled runtimes for the program. Empty list if there are no schedules
    *         or if the program is not found
+   * @throws SchedulerException on unforeseen error.
    */
-  public List<ScheduledRuntime> nextScheduledRuntime(Id.Program program, SchedulableProgramType programType);
+  public List<ScheduledRuntime> nextScheduledRuntime(Id.Program program, SchedulableProgramType programType)
+    throws SchedulerException;
```

Comment:
We can say that in the API, but the implementation actually doesn't check if the program exists or not, we only need to make sure that the app exists, so that we can retrieve the app spec which contains the schedule. Do you think we should also check if the program exists? 

Note that it's also we the endpoint that hits the resume/suspend does, it only checks that the app exists and that the schedule exists in the store, not the program.